hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -168,11 +169,13 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
     String database = identifier.namespace().level(0);
 
     TableOperations ops = newTableOps(identifier);
-    TableMetadata lastMetadata;
-    if (purge && ops.current() != null) {
-      lastMetadata = ops.current();
-    } else {
-      lastMetadata = null;
+    TableMetadata lastMetadata = null;
+    try {
+      if (purge && ops.current() != null) {
+        lastMetadata = ops.current();
+      }
+    } catch (NotFoundException e) {
+      LOG.warn("The metadata file doesn't exist anymore. Continuing to drop the table.", e);
```

Comment:
In Iceberg, log messages should be specific statements and should not speculate about underlying causes. In this case, the operations failed to load table metadata. That doesn't necessarily mean that it doesn't exist. The error should state that clearly. `Failed to load table metadata for table: %s, continuing drop without purge`

Reply:
I have updated log message.